### PR TITLE
Support subscriptions of value types

### PR DESCRIPTION
**Pull request by Shane32**
The hard part was updating all the tests...

### CODE REVIEW

File: src/GraphQL.Tests/Subscription/SubscriptionExecutionStrategyTests.cs [hunk 1]
```diff
@@ -30,6 +30,20 @@ public async Task Basic()
         Observer.ShouldHaveNoMoreResults();
     }
 
+    [Fact]
+    public async Task BasicInt()
```
Comment (Shane32) [inline src/GraphQL.Tests/Subscription/SubscriptionExecutionStrategyTests.cs:34]: This tests the auto-registering types and worked without changes.

File: src/GraphQL.Tests/Subscription/SubscriptionSchema.cs [hunk 2]
```diff
@@ -145,7 +153,7 @@ public MessageInputType()
     {
         Field<StringGraphType>("fromId");
         Field<StringGraphType>("content");
-        Field<DateGraphType>("sentAt");
+        Field<DateTimeOffsetGraphType>("sentAt");
```
Comment (Shane32) [inline src/GraphQL.Tests/Subscription/SubscriptionSchema.cs:156]: So the serialized responses would be repeatable as they would carry timezone info which can be set to UTC.

File: src/GraphQL.Tests/Subscription/SubscriptionTests.cs [hunk 3]
```diff
@@ -127,13 +129,47 @@ public async Task Subscribe()
...
-        message.ShouldBeOfType<ExecutionResult>();
-        message.Data.ShouldNotBeNull();
-        message.Data.ShouldNotBeAssignableTo<Task>();
+        message.ShouldBeSimilarTo("""
+            {"data":{"messageAdded":{"from":{"id":"1","displayName":"test"},"content":"test","sentAt":"2024-01-01T00:00:00\u002B00:00"}}}
+            """);
+    }
+
+    [Fact]
+    public async Task SubscribeInt()
```
Comment (Shane32) [inline src/GraphQL.Tests/Subscription/SubscriptionTests.cs:141]: Tests in this case direct usage of SourceStreamResolver, which is what all the field builders use.

File: src/GraphQL.Tests/Subscription/SubscriptionSchema.cs [hunk 4]
```diff
@@ -70,6 +70,14 @@ public ChatSubscriptions(IChat chat)
             Type = typeof(StringGraphType),
             StreamResolver = new SourceStreamResolver<string>(context => Subscribe(context).Select(message => message.Content))
         });
+
+        int counter = 0;
+        AddField(new FieldType
+        {
+            Name = "messageCounter",
+            Type = typeof(IntGraphType),
+            StreamResolver = new SourceStreamResolver<int>(context => Subscribe(context).Select(_ => ++counter))
```
Comment (Shane32) [inline src/GraphQL.Tests/Subscription/SubscriptionSchema.cs:79]: Threw exception here previously

File: src/GraphQL/Resolvers/SourceStreamResolver.cs [hunk 5]
```diff
@@ -16,9 +16,13 @@ public SourceStreamResolver(Func<IResolveFieldContext, IObservable<TReturnType?>
...
-                throw new InvalidOperationException("The generic type TReturnType must not be a value type.");
-
-            _sourceStreamResolver = context => new ValueTask<IObservable<object?>>((IObservable<object?>)sourceStreamResolver(context));
+            {
+                _sourceStreamResolver = context => new(new ObservableAdapter<TReturnType?>(sourceStreamResolver(context)));
+            }
+            else
+            {
+                _sourceStreamResolver = context => new((IObservable<object?>)sourceStreamResolver(context));
+            }
```
Comment (Shane32) [inline src/GraphQL/Resolvers/SourceStreamResolver.cs:25]: The changes in this file are the only "real" changes in the project

**Review by gao-artur — APPROVED** [review]
Who needs to sleep when you have an OSS to maintain?
That one was fast.

**Comment by Shane32** [comment]
> Who needs to sleep when you have an OSS to maintain? That one was fast.

Yeah, well, my daughter has a fever and I needed to hold her anyway.  I'll be really tired tomorrow.

**Pull request merged**